yksgi/src/lib.rs

```diff
@@ -199,12 +200,36 @@ impl SGInterp {
                     self.switch();
                     continue;
                 }
+                LLVMOpcode::LLVMSub => self.sub(),
+                LLVMOpcode::LLVMSelect => self.select(),
                 _ => todo!("{:?}", self.pc.as_str()),
             }
             self.pc = Value::new(LLVMGetNextInstruction(self.pc.get()));
         }
     }
 
+    fn select(&mut self) {
+        let cond = self.pc.get_operand(0);
+        let op_true = self.pc.get_operand(1);
+        let op_false = self.pc.get_operand(2);
+
+        if op_true.get_type().is_vector() || op_false.get_type().is_vector() {
+            todo!(
+                "vector switches are not implemented: {:?}",
+                self.pc.as_str()
+            );
+        }
+        assert!(cond.get_type().get_int_width() == 1); // Only `i1` is valid.
```

Comment:
I think we should make this a `debug_assert` as this should typically have already been caught by the IR verifier.

Reply:
Oops yes. That was my intention. I've been writing too much C++.

Reply:
Fixed in 8ba0b9c